src/Network/GRPC/MQTT/RemoteClient.hs

```diff
@@ -114,9 +114,25 @@ runRemoteClient ::
   MethodMap ->
   IO ()
 runRemoteClient logger cfg baseTopic methodMap = do
+  runRemoteClient' logger cfg connectMQTT baseTopic methodMap
```

Comment:
If you reorder the arguments so that the new `connect` argument come first, then you can simplify this to `runRemoteClient = runRemoteClient' connectMQTT`. I think that makes it more clear the relationship between `runRemoteClient` and `runRemoteClient'`.

But on that note, I'm not a big fan of using the `'` annotation in the exposed API surface. I would rename it to something more descriptive like `runRemoteClientWithConnect`.